frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -795,9 +812,13 @@ pub mod pallet {
                     contract_id.clone(),
                     staker_reward,
                 ));
+
+                T::Currency::resolve_creating(&staker, reward_imbalance);
+            } else {
+                let delegated_account = DelegatedAccounts::<T>::get(&staker, &contract_id).unwrap_or(staker.clone());
+                T::Currency::resolve_creating(&delegated_account, reward_imbalance);
```

Comment:
Quite elegant.

Does something else needs to be changed/updated related to this call?

Reply:
The Reward event emitted can be changed depending on what account we want to emit in the event, the staker account in wich the rewards has been claimed or the account where the rewards were deposited.

Reply:
Some tools, like Subscan rely on the Reward event to understand how much some staker has earned. Changing the event like this can have wider consequences. Personally I wouldn't do it but it's a good answer to the question.

What about weight required to execute this call?